### PR TITLE
Remove private company markers as appropriate

### LINKED ISSUE #1148: [1846] Remove private markers when appropriate
Currently private markers stay on the board even after they have been used up / removed / phased out, i.e.

After a brown train is purchased or after its power is used, the Lake Shore Line icons should be removed

After a gray train is purchased, the MeatPacking and Steamboats markers are removed

--
Similarly, to make it a bit easier to tell where the MPC is assigned 

After the MeatPacking is assigned to a hex, the marker on the other hex can be removed (e.g. CHI or StL)

### PR DESCRIPTION
**Pull request by benjaminxscott**
closes #1148

### CODE REVIEW

File: lib/engine/game/g_1846.rb [hunk 1]
```diff
@@ -314,8 +314,24 @@ def event_close_companies!
         super
 
         @minors.dup.each { |minor| close_corporation(minor) }
+
+        %w[D14 E17].each do |hex|
```
Comment (tobymao) [inline lib/engine/game/g_1846.rb:318]: docker-compose exec rack rubocop -a 

to auto fix your lint

File: lib/engine/game/g_1846.rb [hunk 2]
```diff
@@ -314,6 +314,23 @@ def event_close_companies!
...
+        end
+      end
+
+      def event_remove_private_tokens!
+        # TODO trigger this
+        %w[B8 C5 D14 G19 I1].each do |hex|
+          hex_by_id(hex).tile.icons.reject! { |icon| icon.name == 'port' }
+        end
+
+        %w[D6 I1].each do |hex|
```
Comment (tobymao) [inline lib/engine/game/g_1846.rb:329]: are there any icons in the game? imo just simplify this, make this one array and just cleare all icons?

Comment (benjaminxscott) [reply lib/engine/game/g_1846.rb:329]: oh yeah that would prob work better

**Comment by benjaminxscott** [comment]
[CI fails](https://github.com/tobymao/18xx/pull/1253/checks?check_run_id=926089784) but I'm not entirely sure if that's due to the test games paying out too much money for steamboats after the bonus was meant to go away

The failures on routing errors, idk

**Comment by tobymao** [comment]
ok i will investigate

**Comment by benjaminxscott** [comment]
I reran it since my local rake was passing w/o error

**Comment by tobymao** [comment]
really? that's strange

**Comment by benjaminxscott** [comment]
This game works, IC is about to buy a gray train and trigger the event

[gray_remove_icons.json.txt](https://github.com/tobymao/18xx/files/4998833/gray_remove_icons.json.txt)

**Pull request merged**